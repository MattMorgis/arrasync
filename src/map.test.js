const {map} = require('./map');

describe('Map test', () => {
  test('first map test', async () => {
    const testArray = [1, 1, 2, 3, 5, 8];
    const mapped = await map(testArray, async num => {
      return num * num;
    });
    expect(mapped).toEqual([1, 1, 4, 9, 25, 64]);
  });

  test('use arguments of mapper function', async () => {
    const testArray = ['a', 'b', 'c', 'd'];
    const mapped = await map(testArray, (value, i, array) => {
      return [value, i, array[0]];
    });
    expect(mapped).toMatchSnapshot();
  });

  test('Should catch async errors', async () => {
    const testArray = ['a', 'b'];
    try {
      await map(testArray, async (item) => {
        try {
          // simulate async error happening
          throw new Error('bang');
        } catch (e) {
          throw e
        }
      });
    } catch (e) {
      expect(e.message).toBe('bang');
    }
    
  });
});
