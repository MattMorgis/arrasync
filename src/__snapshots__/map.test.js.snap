// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Map test use arguments of mapper function 1`] = `
Array [
  Array [
    "a",
    0,
    "a",
  ],
  Array [
    "b",
    1,
    "a",
  ],
  Array [
    "c",
    2,
    "a",
  ],
  Array [
    "d",
    3,
    "a",
  ],
]
`;
